### PR TITLE
[OP#42855] redirect user back to where oauth journey was started

### PR DESCRIPTION
**Pull request by individual-it**
The journey to connect to OpenProject can be started from Dashboar, FilesPage or the personal settings, this PR brings the user back to where the journey was started.

fixes https://community.openproject.org/projects/nextcloud-integration/work_packages/42855

### CODE REVIEW

File: src/components/tab/EmptyContent.vue [hunk 1]
```diff
@@ -15,7 +15,8 @@
 				</div>
 			</div>
 			<div v-if="showConnectButton" class="empty-content--connect-button">
-				<OAuthConnectButton :request-url="requestUrl" />
+				<OAuthConnectButton :request-url="requestUrl"
+					:file-info="fileInfo" />
```
Comment (SwikritiT) [inline src/components/tab/EmptyContent.vue:19]:
```suggestion
				<OAuthConnectButton :request-url="requestUrl" :file-info="fileInfo" />
```
The indent looks a little bit odd

File: src/utils.js [hunk 2]
```diff
@@ -10,6 +13,19 @@ export function delay(callback, ms) {
 	}
 }
 
+export function checkOauthConnectionResult(oauthConnectionResult, oauthConnectionErrorMessage) {
+	if (oauthConnectionResult === 'success') {
+		showSuccess(t('integration_openproject', 'Successfully connected to OpenProject!'))
+	} else if (oauthConnectionResult === 'error') {
```
Comment (SwikritiT) [inline src/utils.js:19]: should this be else?

File: tests/lib/Controller/ConfigControllerTest.php [hunk 3]
```diff
@@ -68,6 +64,33 @@ private function getConfigMock($codeVerifier, $clientSecret) {
...
+	public function testOauthRedirectSuccess():void {
+		$configMock = $this->getConfigMock(
+			str_repeat("A", 128), str_repeat("S", 50));
+		$configMock
+			->method('setUserValue')
+			->withConsecutive(
+				['testUser', 'integration_openproject', 'token', 'oAuthAccessToken'],
+				['testUser', 'integration_openproject', 'refresh_token', 'oAuthRefreshToken'],
+				['testUser', 'integration_openproject', 'user_id', 1],
+				['testUser', 'integration_openproject', 'user_name', 'Tripatti Himal'],
```
Comment (SwikritiT) [inline tests/lib/Controller/ConfigControllerTest.php:84]:
```suggestion
				['testUser', 'integration_openproject', 'user_name', 'Tripathi Himal'],
```
It should be `Tripathi` :wink:

**Review by kiranparajuli589 — COMMENTED** [review]
except for the existing comments, lgtm :+1:

File: tests/lib/Controller/ConfigControllerTest.php (hunk 3 above)
Comment (individual-it) [reply tests/lib/Controller/ConfigControllerTest.php]: good catch! changed!

File: src/utils.js (hunk 2 above)
Comment (individual-it) [reply src/utils.js:19]: it could be empty, in that case we don't want to show anything

File: src/components/tab/EmptyContent.vue (hunk 1 above)
Comment (individual-it) [reply src/components/tab/EmptyContent.vue]: I've put it all in one line, actually its not very long

**Review by kiranparajuli589 — APPROVED** [review]

**Pull request merged**